src/stringobj.c

```diff
@@ -542,14 +546,36 @@ static Obj CopyString(Obj list, Int mut)
 **  characters. The function can be used to print *any* string in a way
 **  which can be read in by GAP afterwards.
 */
-void PrintString(Obj list)
+
+// Type of function given to OutputStringGeneric
+typedef void StringOutputterType(void * data, char * strbuf, UInt len);
+
+// Output using Pr
+void ToPrOutputter(void * data, char * strbuf, UInt len)
+{
+    strbuf[len++] = '\0';
+    Pr("%s", (Int)strbuf, 0L);
+}
+
+// Output to a list of Strings
+void ToStringOutputter(void * data, char * strbuf, UInt len)
+{
+    Obj list = (Obj)data;
+
+    strbuf[len++] = '\0';
+    AddPlist(list, MakeImmString(strbuf));
+    CHANGED_BAG(list);
+}
```

Comment:
To switch from `CHUNK_ESCAPE_STRING` to `VIEW_STRING_FOR_STRING`, I think almost the only thing we would have to do is to change this function to instead append strbuf to a T_STRING given via `data`, i.e.,:
```C
void ToStringOutputter(void * data, const char * buf, UInt lenbuf)
{
    Obj str = (Obj)data;
    Obj lenstr = GET_LEN_STRING(str);
    GROW_STRING(str, lenstr + lenbuf);
    memcpy(CHARS_STRING(str) + lenstr, buf, lenbuf);
    CHARS_STRING(str)[lenstr + lenbuf] = '\0';
}

Reply:
I tried this, but it is producing the wrong answer. I haven't figured out why yet.

Reply:
Well, I forgot the `SET_LEN_STRING` in my sample code... I took the liberty of updating this PR with a commit of mine which makes the proposed changes. If you are happy with them, feel free to squash them into your initial commit